hazelcast/src/main/java/com/hazelcast/map/impl/MapMigrationAwareService.java

```diff
@@ -180,7 +187,9 @@ private void migrateIndex(PartitionMigrationEvent event) {
                     assert event.getNewReplicaIndex() != 0 : "Invalid migration event: " + event;
                     Object value = Records.getValueOrCachedValue(record, serializationService);
                     indexes.removeEntryIndex(key, value);
-                } else if (event.getNewReplicaIndex() == 0) {
+                } else {
+                    assert event.getMigrationEndpoint() == DESTINATION;
+                    assert event.getNewReplicaIndex() == 0;
                     Object value = Records.getValueOrCachedValue(record, serializationService);
```

Comment:
I copied this line from the existing code, but seems like we don't need to mess here with the caching, so this may be simplified, right?